src/kOS.Safe/Encapsulation/Suffixes/ClampSetSuffix.cs

```diff
@@ -16,8 +23,16 @@ public ClampSetSuffix(SuffixGetDlg<TValue> getter, SuffixSetDlg<TValue> setter,
 
         public override void Set(object value)
         {
-            //HACK, this is assumes the value parses as a double 
-            base.Set(Math.Clamp(double.Parse(value.ToString()), min, max));
+            //HACK, this is assumes the value parses as a double
+            var dblValue = double.Parse(value.ToString());
+            
+            //Rounding to the nearest step
+            if (!(System.Math.Abs(stepIncrement) < 0.00001))
+            {
+                dblValue = System.Math.Round(dblValue/stepIncrement)*stepIncrement;
+            }
```

Comment:
This may be a weird case that will just never come up, but if your min/max bounds are not exact multiples of the increment, this can cause the value to become rounded to a value that is outside the bounds.
For example: You want the valid values to be the integers 9,13,17,21,25.  So you set a min value of 9, a max value of 25, and an increment of 4, thinking it will give you values (min, min+inc, min+2*inc, min+3*inc, min+4*inc).  But what you have doesn't do that.  It gives you the values 8,12,16,22,26.  The rounding allows it to go out of the range bounds.  (i.e. a value of 9.01 gets rounded down to 8).

I hit a similar problem when trying to enforce the range increments in KSPField tweakables, and had to write something more complex to deal with what otherwise seems like a simple thing.  (You can see the code over in the PartModuleFields.cs file.)